ghmirror/data_structures/requests_cache.py

```diff
@@ -0,0 +1,64 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Maha Ashour <mashour@redhat.com>
+
+"""
+Implements caching backend
+"""
+
+import os
+from abc import abstractmethod
+
+from ghmirror.data_structures.monostate import InMemoryCache
+from ghmirror.data_structures.redis_data_structures import RedisCache
+
+
+class RequestsCache:
+    """
+    Inherits either InMemoryCache or Redis Cache object
+    """
+    def __new__(cls, *args, **kwargs):
+        cache_type = os.environ.get('GITHUB_MIRROR_CACHE_TYPE', 'in-memory')
+        if cache_type == 'redis':
+            return RedisCache(*args, **kwargs)
+        return InMemoryCache(*args, **kwargs)
+
+    @abstractmethod
+    def __init__(self):
+        """
+        Intialization
```

Comment:
those docstrings could me more useful... otherwise, you can just use `pass` instead.

Reply:
ok will use pass instead